yourbench/utils/configuration_engine.py

```diff
@@ -220,9 +226,35 @@ def set_default_provider(self):
 
     @model_validator(mode="after")
     def validate_api_keys(self):
```

Comment:
Isn't the content of this function redundant with the contents of main? Could you do an import to avoid duplication?